File: Employee.List.App.Client/EmployeeListApp.Client/EmployeeListApp.Domain/Entities/Employee.cs
using EmployeeListApp.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace EmployeeListApp.Domain.Entities
{
    public class Employee
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int EmployeeId { get; set; }

        [Required]
        [MaxLength(100)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string LastName { get; set; } = string.Empty;

        [MaxLength(100)]
        public string? MiddleName { get; set; }

        [MaxLength(200)]
        public string Email { get; set; } = string.Empty;

        [MaxLength(250)]
        public string Address { get; set; } = string.Empty;

        [MaxLength(50)]
        public string PhoneNumber { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Salary { get; set; }

        public EmployeeStatus Status { get; set; }
        public EmployeeGender Gender { get; set; }
    }
}

File: employee-list-app.UI/src/app/interfaces/employee.ts
import { EmployeeStatus } from '../enums/employee-status.enum';
import { EmployeeGender } from '../enums/employee-gender.enum';

export interface IEmployee {
  employeeId: number;
  firstName: string;
  lastName: string;
  middleName: string;
  email: string;
  address: string;
  phoneNumber: string;
  salary: number;
  status: EmployeeStatus;
  gender: EmployeeGender;
}
